File: Arcade/Intro/21_isIPv4Address.js
function isIPv4Address(inputString) {
    
    var array = inputString.split('.');
    if (array.length !== 4) { return false; }
    
    var pattern = new RegExp("[^0-9]");
    
    for (var i = 0; i < array.length; i++) {
        if (array[i] > 255 || array[i] < 0 || array[i] === '') {
            return false;
        } else if (pattern.test(array[i])) {
            return false;
        }
    } return true;
}

File: Arcade/Intro/C++/21_isIPv4Address.cpp
bool isIPv4Address(std::string inputString) {
    vector<string> temp;
    stringstream s(inputString);
    while(!s.eof()){
        temp.emplace_back();
        getline(s,temp.back(),'.');
    }
    if(temp.size()==4){
        for(string i:temp){
            for(char c:i){
                if(c<'0'||c>'9') return false;
            }
            int n=atoi(i.c_str());
            if(n<0 || n>255) return false;
            if(i=="") return false;
        }
        return true;
    }
    return false;
}
